File: hw9/nodejs/mysql_test_1.js

var mysql = require('mysql');
const config = require('./config.json');

// create the connection
var con = mysql.createConnection(config);

// basic query example
var query = 'SELECT * FROM category ORDER BY name';

// open the connection
con.connect(function(err) {
    if (err) throw err;
    console.log('>>> Connected to the database');
});

// run the query and log to output
con.query(query, function(err, result, fields) {
    if (err) throw err;
    for (const r of result) {
        console.log(r['name']);
    }
});

// close the connection
con.end(function(err) {
    if (err) throw err;
    console.log('>>> Connection to the database closed');
});

File: hw8/python/mysql_test_1.py
import mysql.connector
import config

def main():
    try: 
        # connection info
        usr = config.mysql['user']
        pwd = config.mysql['password']
        hst = config.mysql['host']
        dab = 'cpsc321'
        # create a connection
        con = mysql.connector.connect(user=usr,password=pwd, host=hst,
                                      database=dab)
        # create a result set
        rs = con.cursor()
        query = 'SELECT category_id, name from category'
        # execute the query
        rs.execute(query)
        # print the results
        for (catid, cname) in rs:
            print ("{}, {}".format(catid, cname))

        rs.close()
        con.close()
        
    except mysql.connector.Error as err:
        print (err)


if __name__ == '__main__':
    main()
    
